pkg/repository/postgres/tenant.go

```diff
@@ -35,7 +36,15 @@ func NewTenantAPIRepository(shared *sharedRepository, cache cache.Cacheable, def
 	}
 }
 
-func (r *tenantAPIRepository) CreateTenant(ctx context.Context, opts *repository.CreateTenantOpts) (*dbsqlc.Tenant, error) {
+func (r *tenantAPIRepository) RegisterCreateCallback(callback repository.UnscopedCallback[*dbsqlc.Tenant]) {
+	if r.createCallbacks == nil {
+		r.createCallbacks = make([]repository.UnscopedCallback[*dbsqlc.Tenant], 0)
+	}
+
+	r.createCallbacks = append(r.createCallbacks, callback)
+}
```

Comment:
O_O does Go allow this? Haven't seen this pattern before so I'm curious about it, since this feels a lot more like Java getter/setter patterns to me than Go's no-class approach to things